[Toh Chooi Ern, Jasmine] Duke Increments

tohcejasmine opened the pull request:
Code updates till Week 3

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

tohcejasmine committed:
add .gitignore file

tohcejasmine committed:
level 1: greet, echo, exit

tohcejasmine committed:
level 2: add, list (able to add, store and list tasks)

tohcejasmine committed:
level 3: mark as done

tohcejasmine committed:
Update .gitignore

tohcejasmine committed:
Add in extension class Task

tohcejasmine committed:
Update to ignore all .class files

tohcejasmine committed:
level 4: todos, events, deadlines (add support for tracking 3 types of tasks

tohcejasmine committed:
Add in new classes ToDo, Deadline, Event

tohcejasmine committed:
Merge branch 'gradle'

tohcejasmine committed:
Update Gradle build file

tohcejasmine committed:
A-TextUiTesting: test using I/O redirection technique

Changes to be committed:
	new file:   text-ui-test/EXPECTED.txt
	new file:   text-ui-test/input.text
	new file:   text-ui-test/runtest.bat

* Note: Currently, running batch file, unable to compile Duke.java and create
ACTUAL.TXT

tohcejasmine committed:
Restructure approach to involve ArrayList instead of Scanner methods

tohcejasmine committed:
Level 5: Handle errors (incorrect inputs by users)

tohcejasmine committed:
Upload Exception classes

tohcejasmine committed:
Level 6: Add support for deleting tasks from list

tohcejasmine committed:
Neaten up code

* Single whitespace after comment //
* Abstract out repeated codes into their own methods
* Neaten up the code for printing to console
* Standardise code for switch statement to match coding standards in
class (no indentation after each case)

tohcejasmine committed:
Properly indent switch statement

tohcejasmine committed:
Update .gitignore to ignore task list saved in a txt file

tohcejasmine committed:
Level-7: Save tasks in hard disk automatically

 On branch branch-Level-7
 Changes to be committed:
	modified:   src/main/java/Duke.java
	new file:   src/main/java/DukeData.java
	modified:   src/main/java/DukeException.java
	modified:   src/main/java/Task.java

tohcejasmine committed:
Level 8: Recognise and parse descriptions of date-times of format dd/MM/yyyy HHmm for deadline tasks

tohcejasmine committed:
Merge branch 'branch-Level-7'

tohcejasmine committed:
Merge branch 'branch-Level-8'

* Resolve merge conflicts by incorporating new methods

tohcejasmine committed:
Shift Level 8 logic of interpreting date-time to be handled by Deadline class instead of main class

tohcejasmine committed:
A-MoreOOP: Extract out closely related code as classes

* Ui
* Storage (modified from previously named DukeData)
* Parser
* TaskList
* Command

Changes to be committed:
	new file:   src/main/java/Command.java
	modified:   src/main/java/Duke.java
	modified:   src/main/java/DukeException.java
	new file:   src/main/java/Parser.java
	renamed:    src/main/java/DukeData.java -> src/main/java/Storage.java
	new file:   src/main/java/TaskList.java
	new file:   src/main/java/Ui.java

tohcejasmine committed:
Upload basic test files, actual methods not written yet

Changes to be committed:
	new file:   src/test/java/DukeTest.java
	new file:   src/test/java/ParserTest.java
	new file:   src/test/java/StorageTest.java
	new file:   src/test/java/TaskListTest.java
	new file:   src/test/java/UiTest.java

tohcejasmine committed:
A-JUnit: Add test methods for Parser and TaskList

* Also added helper methods in Duke and Command classes to help with
debugging
* print() and equals()
* Note: Don't think equals() methods are properly defined to work yet

Changes to be committed:
	modified:   src/main/java/Command.java
	modified:   src/main/java/Duke.java
	modified:   src/test/java/ParserTest.java
	modified:   src/test/java/StorageTest.java
	modified:   src/test/java/TaskListTest.java
	modified:   src/test/java/UiTest.java

tohcejasmine committed:
A-JavaDoc: add JavaDoc comments to the code

On branch A-JavaDoc
 Changes to be committed:
	modified:   src/main/java/Command.java
	modified:   src/main/java/Duke.java
	modified:   src/main/java/DukeException.java
	modified:   src/main/java/Storage.java
	modified:   src/main/java/Task.java
	modified:   src/main/java/TaskList.java
	modified:   src/main/java/Ui.java

tohcejasmine committed:
A-CodingStandard: Tweak code to comply with coding standard

* As specified in course website

On branch branch-A-CodingStandard
 Changes to be committed:
	modified:   src/main/java/Command.java
	modified:   src/main/java/Storage.java
	modified:   src/main/java/Task.java
	modified:   src/main/java/TaskList.java
	modified:   src/main/java/Ui.java

tohcejasmine committed:
Level 9: Find a task by keyword

On branch branch-Level-9
 Changes to be committed:
	modified:   src/main/java/Command.java
	modified:   src/main/java/Parser.java
	modified:   src/main/java/Task.java
	modified:   src/main/java/Ui.java

tohcejasmine committed:
A-JavaDoc: Merge branch 'branch-A-JavaDoc'

tohcejasmine committed:
Merge branch 'branch-A-CodingStandard'

 Conflicts:
	src/main/java/Task.java
* Fix conflict by incorporating both additions

tohcejasmine committed:
Level 9: Merge branch 'branch-Level-9'

Conflicts:
	src/main/java/Command.java
	src/main/java/Task.java
	src/main/java/Ui.java

* Incorporate additions from both branches
* Accept changes in Task methods from Level 9

tohcejasmine committed:
Resolve some checkStyle violations

tohcejasmine committed:
JavaFX Part 1

On branch branch-Level-10

* Also resolved some checkStyle violations

Changes to be committed:
	new file:   AddCommand.java
	modified:   Command.java
	new file:   Deadline.java
	new file:   DeleteCommand.java
	new file:   DoneCommand.java
	modified:   Duke.java
	new file:   DukeDataException.java
	modified:   DukeException.java
	new file:   EmptyDescriptionException.java
	new file:   Event.java
	new file:   ExitCommand.java
	new file:   FindCommand.java
	new file:   IncorrectInfoInputException.java
	new file:   Launcher.java
	new file:   ListCommand.java
	new file:   SubCommand.java
	modified:   Task.java
	new file:   ToDo.java
	new file:   UnknownCommandException.java

tohcejasmine committed:
Update: JavaFX Part 1

tohcejasmine committed:
Level 10: JavaFX Part 2

tohcejasmine committed:
Level 10: In between JavaFX Part 3 and Part 4

tohcejasmine committed:
Level 10: JavaFX Part 4

tohcejasmine committed:
Also Level 10: JavaFX Part 4

tohcejasmine committed:
Add in markers for Level 10: JavaFX portions

tohcejasmine committed:
Merge branch 'branch-Level-10'

AHaliq commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,44 @@
+        while (inProgram) {
+            String userCommand = input.next();
+            if (userCommand.equals("bye")) {
+                System.out.println("-----");
```
~~you could create a function that takes a string as argument and that prints it out with the horizontal lines instead of repeatedly printing them before and after every output e.g:~~ Just saw your next commit with the display function.

```Java
public void formatPrint(String str) {
    System.out.println("-----");
    System.out.println(str);
    System.out.println("-----");
}
```

AHaliq commented on src/main/java/Duke.java:
```diff
@@ -19,26 +24,45 @@ public static void main(String[] args) {
-                System.out.println("    " + userCommand);
-                System.out.println("-----");
+            String userCommand = input.nextLine();
+            switch (userCommand) {
```
Could use an enum here

AHaliq commented on src/main/java/Duke.java:
```diff
@@ -64,5 +65,27 @@ private void display(String response) {
     }
 
+    //List out added tasks
+    private void displayTasks() {
```
good approach to extract into its own function!

AHaliq commented on src/main/java/Duke.java:
```diff
@@ -64,5 +65,27 @@ private void display(String response) {
 
+    //List out added tasks
+    private void displayTasks() {
+        System.out.println("-----");
```
could have used your display function instead of printing out ----- again.

AHaliq commented on src/main/java/Duke.java:
```diff
@@ -64,5 +65,27 @@ private void display(String response) {
+    private void displayMarkDone(int doneIdx) {
+        Task doneTask = taskList.get(doneIdx-1);
+        doneTask.markDone();
+        System.out.println("-----");
```
same here, consider utilising your display function

AHaliq commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,22 @@
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
```
`this.attribute` (for example `this.description`) or `attribute` (for example `return (isDone?...`), either way is fine, but stick to one convention

AHaliq commented on src/main/java/Duke.java:
```diff
@@ -46,46 +49,109 @@ private void run() {
+                            indentString +
+                            "List of valid commands: \n" +
+                            indentString +
+                            "todo, deadline, event, list, done, bye");
```
nice touch guiding the users to the right commands!

AHaliq commented on src/main/java/Duke.java:
```diff
@@ -50,45 +62,71 @@ private void run() {
+                        int lastByIdx = userInputArr.lastIndexOf(subCommand);
+                        //No "/by" or multiple "/by"s provided
+                        if (firstByIdx == -1 || firstByIdx != lastByIdx) {
+                            throw new IncorrectInfoInputException(subCommand);
```
good use of exceptions!

AHaliq reviewed (commented):
A good straight forward approach using switch statements for command types.

cambrian-dk commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,132 @@
+    // Constructor
+    public Parser() { }
+
+    public Command parse(String userInput, TaskList taskList) throws DukeException {
```
Maybe a comment here would be useful to explain what the parse method does generally, on top of the comments within the method

cambrian-dk commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,132 @@
+            } else {
+                // Check if integer is provided
+                try {
+                    int deleteIdx = Integer.parseInt(userInputArr.get(1));
```
Maybe it would reduce nesting if the logic of the commands was refactored into the class files of those commands.

cambrian-dk reviewed (commented):
Good coverage of incorrect usage with exceptions, some design choices can be made to make the code more SLAP(but that's a subjective choice)

tohcejasmine committed:
In progress: cleaning up code

tohcejasmine committed:
Code-Quality: Restructure and improve code quality

* Rollback JavaFx methods and changes
* Restructure Ui.java
* Restructure Parser.java

 Changes to be committed:
	modified:   src/main/java/DoneCommand.java
	modified:   src/main/java/Duke.java
	modified:   src/main/java/FindCommand.java
	modified:   src/main/java/Parser.java
	modified:   src/main/java/Ui.java

tohcejasmine committed:
Trying to improve code quality

* Code currently broken :-(

tohcejasmine committed:
Allow GUI to understand Duke commands

* Modify execute in Command classes to return String
* Modify Ui methods to return String instead of void

tohcejasmine committed:
Improve code structure

tohcejasmine committed:
Move classes to packages

tohcejasmine committed:
Improve code quality

* Ui: Abstract out methods
* Ui: Include multiple methods of same name but different signatures
* DukeException: Create child classes used in different classes
* StaticString: Consolidate messages and responses used in this class
* Command: Change execute() to take in TaskList and Storage instead of
Duke itself

tohcejasmine committed:
Handle some not so common cases

* Notify user if task has already been marked done
* Handle no subcommand description (bug in logic)
* Add in note of SimpleDateFormat (yy AD)
* Change Find command to not be case sensitive to keywords

tohcejasmine committed:
Merge branch 'branch-A-CodeQuality'

* Accept version from branch over master

tohcejasmine committed:
JavaFX Part 1

* Note: DukeGUI constructor has to take in no arguments
* Shifted file path to within the constructor

tohcejasmine committed:
JavaFX Part 2 - Step 1

tohcejasmine committed:
JavaFX Part 2 - Step 2

tohcejasmine committed:
JavaFX Part 3 - Iteration 1

tohcejasmine committed:
JavaFX Part 3 - Iteration 1 (update method to allow ScrollPane to scroll down automatically

tohcejasmine committed:
JavaFX Part 3 - Iteration 2

tohcejasmine committed:
JavaFX Part 3 - Iteration 3

tohcejasmine committed:
A-Assertions: add in assert statements

tohcejasmine committed:
Merge pull request #6 from tohcejasmine/branch-A-Assertions

A-Assertions: add in assert statements

tohcejasmine committed:
Refactor some code using streams and lambdas

tohcejasmine committed:
Merge pull request #7 from tohcejasmine/branch-A-Streams

A-Streams, A-Lambdas: Refactor some code using streams and lambdas

tohcejasmine committed:
Add in todos for extension

tohcejasmine committed:
Add to C-Statistics

* Add log file after each successful execution of command
* Class to read and save log file
* Class to represent a log

tohcejasmine committed:
Complete methods to show stats on console

* Complete execute() of stats command to display numbers
* Add static strings used in display
* Add methods to get numbers

tohcejasmine committed:
Merge pull request #8 from tohcejasmine/branch-C-Statistics

Branch C-Statistics

tohcejasmine committed:
Merge branch 'master' into branch-GUI

tohcejasmine committed:
JavaFX Part 4

tohcejasmine committed:
Update build.gradle

tohcejasmine committed:
Merge branch 'branch-GUI'

tohcejasmine committed:
Credit approach to showing GUI response to input

tohcejasmine committed:
Add in statement to close JavaFX window when 'bye' command entered

tohcejasmine committed:
Make product name visible in title bar of Window

tohcejasmine committed:
Add user guide

tohcejasmine committed:
Add imports in test classes

tohcejasmine committed:
Add dependencies for cross-platforms

tohcejasmine committed:
Change UI (ImageView, Label)

tohcejasmine committed:
Update Gradle file

tohcejasmine committed:
Change background image of GUI to be more opaque

tohcejasmine committed:
Set theme jekyll-theme-modernist

tohcejasmine committed:
Update Ui.png

tohcejasmine committed:
Merge branch 'master' of github.com:tohcejasmine/duke

tohcejasmine committed:
Remove redundant imports

tohcejasmine committed:
Update build.gradle

tohcejasmine committed:
Update README.md

tohcejasmine committed:
Improve code and Fix bugs

* Shift filepath to be entered on Storage side
* Let Event tasks recognised date-time of specific format as well
* Fix bug in Stat

tohcejasmine committed:
Update README.md

tohcejasmine committed:
Update README.md

tohcejasmine committed:
Fix exception thrown, exception message

tohcejasmine committed:
Generalise file saving of data to whichever place application is run

tohcejasmine committed:
Update README.md

tohcejasmine committed:
Update build.gradle

tohcejasmine committed:
Generalise saving of storage and filepaths

* Save to \data folder if running on @tohcejasmine local repo
* Create and save to \dukeData folder in working dir if running externally (e.g. jar file)

tohcejasmine committed:
Upload Travis CI file

tohcejasmine committed:
Give Travis permission access in runtime execution

tohcejasmine committed:
Update README

* Add Travis CI badge
